### PR TITLE
Packaging for elastic agent

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

This PR introduces a `mage package` command. 
Result of this mage package command is an archive consisting of
- binary
- spec file
- config file

This archive is then consumed by elastic-agent and included in agent release artifact

## Why is it important?

For agent to be able to work with shipper

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.md` or `CHANGELOG-developer.md`.

### CODE REVIEW

**Review by leehinman — APPROVED** [review]
be good to see if @faec has input before merging

File: elastic-agent-shipper.spec.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      - darwin/amd64
+      - darwin/arm64
+      - windows/amd64
+      - container/amd64
+      - container/arm64
+    command: {}
+  - name: logstash
+    description: Logstash
+    platforms: *platforms
+    command: {}
```
Comment (cmacknz) [inline elastic-agent-shipper.spec.yml:17]:
```suggestion
```

We should avoid including this until we actually implement a Logstash output: https://github.com/elastic/elastic-agent-shipper/issues/129

The Elasticsearch output is already in progress at least, so we can keep that one.

File: magefile.go [hunk 2]
```diff
@@ -215,3 +220,227 @@ func Notice() error {
...
+
+		if err != nil {
+			panic(err)
+		}
+
+		// generate sha sum
+		archiveFullPath := filepath.Join(archivePath, archiveFileName)
+		archive, err := os.Open(archiveFullPath)
+		if err != nil {
+			panic(errors.Wrapf(err, "failed opening archive %q", archiveFileName))
```
[the remark this answers is not in the document]
Comment (michalpristas) [reply magefile.go:293]: yep

File: elastic-agent-shipper.spec.yml (hunk 1 above)
Comment (michalpristas) [reply elastic-agent-shipper.spec.yml:17]: makes sense, thanks

**Review by cmacknz — APPROVED** [review]
Confirmed the `mage package` target works locally for me, thanks!

**Pull request merged**